### ISSUE TITLE
GTFS-rt - Wrong stop_ids in trip update

### ISSUE THREAD

**Issue by cagryInside** [CUTR-at-USF/psta-data#10]
Gtfs-rt data returns the following for `trip_id: 1208463020`:

```
entity {
  id: "53"
  trip_update {
    trip {
      trip_id: "1208463020"
      route_id: "CAT"
    }
    stop_time_update {
      departure {
        time: 1435257362
      }
      stop_id: "4995"
    }
    stop_time_update {
      departure {
        time: 1435257362
      }
      stop_id: "5453"
    }
    stop_time_update {
      arrival {
        time: 1435257393
      }
      stop_id: "1605"
    }
    stop_time_update {
      arrival {
        time: 1435257498
      }
      stop_id: "1606"
    }
    stop_time_update {
      arrival {
        time: 1435257547
      }
      stop_id: "1607"
    }
    stop_time_update {
      arrival {
        time: 1435257633
      }
      stop_id: "974"
    }
    stop_time_update {
      arrival {
        time: 1435257666
      }
      stop_id: "975"
    }
...
```

However, in the static Gtfs data, this trip defined as following:

```
1208463020,14:35:00,14:35:00,5453,1,,0,0, 
1208463020,,,1605,2,,0,0, 
1208463020,,,1606,3,,0,0, 
1208463020,,,1607,4,,0,0, 
1208463020,,,974,5,,0,0, 
1208463020,,,975,6,,0,0, 
1208463020,,,976,7,,0,0, 
1208463020,,,977,8,,0,0, 
1208463020,,,1608,9,,0,0, 
1208463020,,,1609,10,,0,0, 
1208463020,,,1610,11,,0,0, 
1208463020,,,1611,12,,0,0, 
1208463020,,,1612,13,,0,0, 
1208463020,,,1613,14,,0,0, 
1208463020,,,1614,15,,0,0, 
1208463020,,,1615,16,,0,0, 
1208463020,,,1616,17,,0,0, 
1208463020,,,1617,18,,0,0, 
1208463020,,,1618,19,,0,0, 
1208463020,,,1619,20,,0,0, 
1208463020,,,1620,21,,0,0, 
1208463020,,,1621,22,,0,0, 
1208463020,,,1622,23,,0,0, 
1208463020,,,1623,24,,0,0, 
1208463020,,,1624,25,,0,0, 
1208463020,,,1625,26,,0,0, 
1208463020,,,1626,27,,0,0, 
1208463020,,,1627,28,,0,0, 
1208463020,,,1628,29,,0,0, 
1208463020,,,1629,30,,0,0, 
1208463020,,,1630,31,,0,0, 
1208463020,,,1631,32,,0,0, 
1208463020,,,1632,33,,0,0, 
1208463020,,,1633,34,,0,0, 
1208463020,,,1634,35,,0,0, 
1208463020,,,5167,36,,0,0, 
1208463020,,,4102,37,,0,0, 
1208463020,,,4103,38,,0,0, 
1208463020,,,4104,39,,0,0, 
1208463020,,,5176,40,,0,0, 
1208463020,,,4105,41,,0,0, 
1208463020,,,4106,42,,0,0, 
1208463020,,,4107,43,,0,0, 
1208463020,,,9927,44,,0,0, 
1208463020,,,5169,45,,0,0, 
1208463020,,,9942,46,,0,0, 
1208463020,,,9939,47,,0,0, 
1208463020,,,5171,48,,0,0, 
1208463020,,,9930,49,,0,0, 
1208463020,,,3292,50,,0,0, 
1208463020,,,1711,51,,0,0, 
1208463020,,,5200,52,,0,0, 
1208463020,,,5201,53,,0,0, 
1208463020,15:20:00,15:20:00,5202,54,,0,0, 
```

So, I think first two stop time updates are wrong in the Gtfs-rt.

cc @barbeau

**Comment by barbeau** [comment]
@cagryInside Agreed, looks like `stop_id: "4995"` shouldn't be in the GTFS-rt feed.  I'll tell CD.  Thanks for finding this!

Also - do you have a timestamp for when this was generated, so they can troubleshoot in their logs?

**Comment by cagryInside** [comment]
Here is the timestamp and vehicle_id:

```
    vehicle {
      id: "925"
    }
    timestamp: 1435256576

```

**Comment by cagryInside** [comment]
I found another error for `trip_id: 1208461020` - again, `stop_id 4995` shouldn't be included in this real-time data because it isn't visited by this trip:

```
entity {
  id: "120"
  trip_update {
    trip {
      trip_id: "1208461020"
      route_id: "CAT"
    }
    stop_time_update {
      departure {
        time: 1435687797
      }
      stop_id: "4995"
    }
    stop_time_update {
      departure {
        time: 1435687797
      }
      stop_id: "5453"
    }
    stop_time_update {
      arrival {
        time: 1435687828
      }
      stop_id: "1605"
    }
    stop_time_update {
      arrival {
        time: 1435687933
      }
      stop_id: "1606"
    }
    stop_time_update {
      arrival {
        time: 1435687982
      }
      stop_id: "1607"
    }
    stop_time_update {
      arrival {
        time: 1435688068
      }
      stop_id: "974"
    }
    stop_time_update {
      arrival {
        time: 1435688101
      }
      stop_id: "975"
    }
    stop_time_update {
      arrival {
        time: 1435688148
      }
      stop_id: "976"
    }
    stop_time_update {
      arrival {
        time: 1435688181
      }
      stop_id: "977"
    }
    stop_time_update {
      arrival {
        time: 1435688220
      }
      stop_id: "1608"
    }
    stop_time_update {
      arrival {
        time: 1435688272
      }
      stop_id: "1609"
    }
    stop_time_update {
      arrival {
        time: 1435688299
      }
      stop_id: "1610"
    }
    stop_time_update {
      arrival {
        time: 1435688323
      }
      stop_id: "1611"
    }
    stop_time_update {
      arrival {
        time: 1435688351
      }
      stop_id: "1612"
    }
    stop_time_update {
      arrival {
        time: 1435688413
      }
      stop_id: "1613"
    }
    stop_time_update {
      arrival {
        time: 1435688463
      }
      stop_id: "1614"
    }
    stop_time_update {
      arrival {
        time: 1435688493
      }
      stop_id: "1615"
    }
    stop_time_update {
      arrival {
        time: 1435688526
      }
      stop_id: "1616"
    }
    stop_time_update {
      arrival {
        time: 1435688556
      }
      stop_id: "1617"
    }
    stop_time_update {
      arrival {
        time: 1435688589
      }
      stop_id: "1618"
    }
    stop_time_update {
      arrival {
        time: 1435688620
      }
      stop_id: "1619"
    }
    stop_time_update {
      arrival {
        time: 1435688653
      }
      stop_id: "1620"
    }
    stop_time_update {
      arrival {
        time: 1435688715
      }
      stop_id: "1621"
    }
    stop_time_update {
      arrival {
        time: 1435688755
      }
      stop_id: "1622"
    }
    stop_time_update {
      arrival {
        time: 1435688784
      }
      stop_id: "1623"
    }
    stop_time_update {
      arrival {
        time: 1435688826
      }
      stop_id: "1624"
    }
    stop_time_update {
      arrival {
        time: 1435688865
      }
      stop_id: "1625"
    }
    stop_time_update {
      arrival {
        time: 1435688909
      }
      stop_id: "1626"
    }
    stop_time_update {
      arrival {
        time: 1435688925
      }
      stop_id: "1627"
    }
    stop_time_update {
      arrival {
        time: 1435688968
      }
      stop_id: "1628"
    }
    stop_time_update {
      arrival {
        time: 1435689017
      }
      stop_id: "1629"
    }
    stop_time_update {
      arrival {
        time: 1435689047
      }
      stop_id: "1630"
    }
    stop_time_update {
      arrival {
        time: 1435689096
      }
      stop_id: "1631"
    }
    stop_time_update {
      arrival {
        time: 1435689129
      }
      stop_id: "1632"
    }
    stop_time_update {
      arrival {
        time: 1435689166
      }
      stop_id: "1633"
    }
    stop_time_update {
      arrival {
        time: 1435689192
      }
      stop_id: "1634"
    }
    stop_time_update {
      arrival {
        time: 1435689270
      }
      stop_id: "5167"
    }
    stop_time_update {
      arrival {
        time: 1435689360
      }
      stop_id: "4102"
    }
    stop_time_update {
      arrival {
        time: 1435689417
      }
      stop_id: "4103"
    }
    stop_time_update {
      arrival {
        time: 1435689472
      }
      stop_id: "4104"
    }
    stop_time_update {
      arrival {
        time: 1435689499
      }
      stop_id: "5176"
    }
    stop_time_update {
      arrival {
        time: 1435689528
      }
      stop_id: "4105"
    }
    vehicle {
      id: "822"
    }
    timestamp: 1435687762
  }
}

```

Here is the gtfs data for this trip:

```
1208461020,14:10:00,14:10:00,5453,1,,0,0, 
1208461020,,,1605,2,,0,0, 
1208461020,,,1606,3,,0,0, 
1208461020,,,1607,4,,0,0, 
1208461020,,,974,5,,0,0, 
1208461020,,,975,6,,0,0, 
1208461020,,,976,7,,0,0, 
1208461020,,,977,8,,0,0, 
1208461020,,,1608,9,,0,0, 
1208461020,,,1609,10,,0,0, 
1208461020,,,1610,11,,0,0, 
1208461020,,,1611,12,,0,0, 
1208461020,,,1612,13,,0,0, 
1208461020,,,1613,14,,0,0, 
1208461020,,,1614,15,,0,0, 
1208461020,,,1615,16,,0,0, 
1208461020,,,1616,17,,0,0, 
1208461020,,,1617,18,,0,0, 
1208461020,,,1618,19,,0,0, 
1208461020,,,1619,20,,0,0, 
1208461020,,,1620,21,,0,0, 
1208461020,,,1621,22,,0,0, 
1208461020,,,1622,23,,0,0, 
1208461020,,,1623,24,,0,0, 
1208461020,,,1624,25,,0,0, 
1208461020,,,1625,26,,0,0, 
1208461020,,,1626,27,,0,0, 
1208461020,,,1627,28,,0,0, 
1208461020,,,1628,29,,0,0, 
1208461020,,,1629,30,,0,0, 
1208461020,,,1630,31,,0,0, 
1208461020,,,1631,32,,0,0, 
1208461020,,,1632,33,,0,0, 
1208461020,,,1633,34,,0,0, 
1208461020,,,1634,35,,0,0, 
1208461020,,,5167,36,,0,0, 
1208461020,,,4102,37,,0,0, 
1208461020,,,4103,38,,0,0, 
1208461020,,,4104,39,,0,0, 
1208461020,,,5176,40,,0,0, 
1208461020,,,4105,41,,0,0, 
1208461020,,,4106,42,,0,0, 
1208461020,,,4107,43,,0,0, 
1208461020,,,9927,44,,0,0, 
1208461020,,,5169,45,,0,0, 
1208461020,,,9942,46,,0,0, 
1208461020,,,9939,47,,0,0, 
1208461020,,,5171,48,,0,0, 
1208461020,,,9930,49,,0,0, 
1208461020,14:50:00,14:50:00,3292,50,,0,0, 

```

You can also find all gtfs-rt data for this moment:
https://gist.github.com/cagryInside/46d51b42f5407d4567f1

**Comment by cagryInside** [comment]
Same problem still exist for `trip_id: 1208463020`
Gtfs-rt and gtfs: https://gist.github.com/cagryInside/c96476c44334ea64b699

**Comment by cagryInside** [comment]
Found the same problem for different trip (`trip_id = 1209104020`) - 2nd entry `stop_id 5453` isn't in the GTFS data.

Gtfs-rt:

```
entity {
  id: "78"
  trip_update {
    trip {
      trip_id: "1209104020"
      route_id: "SBT"
    }
    stop_time_update {
      departure {
        time: 1436918098
      }
      stop_id: "4995"
    }
    stop_time_update {
      departure {
        time: 1436918098
      }
      stop_id: "5453"  // <-- isn’t in GTFS data for this trip
    }
    stop_time_update {
      arrival {
        time: 1436918132
      }
      stop_id: "4996"
    }
    stop_time_update {
      arrival {
        time: 1436918162
      }
      stop_id: "4997"
    }
    stop_time_update {
      arrival {
        time: 1436918191
      }
      stop_id: "4998"
    }
    stop_time_update {
      arrival {
        time: 1436918200
      }
      stop_id: "4999"
    }
    stop_time_update {
      arrival {
        time: 1436918229
      }
      stop_id: "5000"
    }
    stop_time_update {
      arrival {
        time: 1436918249
      }
      stop_id: "5001"
    }
    stop_time_update {
      arrival {
        time: 1436918280
      }
      stop_id: "5002"
    }
    stop_time_update {
      arrival {
        time: 1436918338
      }
      stop_id: "5003"
    }
    stop_time_update {
      arrival {
        time: 1436918379
      }
      stop_id: "5004"
    }
    stop_time_update {
      arrival {
        time: 1436918432
      }
      stop_id: "5005"
    }
    stop_time_update {
      arrival {
        time: 1436918459
      }
      stop_id: "5006"
    }
    stop_time_update {
      arrival {
        time: 1436918495
      }
      stop_id: "9933"
    }
    stop_time_update {
      arrival {
        time: 1436918519
      }
      stop_id: "5007"
    }
    stop_time_update {
      arrival {
        time: 1436918561
      }
      stop_id: "5009"
    }
    stop_time_update {
      arrival {
        time: 1436918593
      }
      stop_id: "5010"
    }
    stop_time_update {
      arrival {
        time: 1436918638
      }
      stop_id: "5011"
    }
    stop_time_update {
      arrival {
        time: 1436918689
      }
      stop_id: "5012"
    }
    stop_time_update {
      arrival {
        time: 1436918726
      }
      stop_id: "5013"
    }
    stop_time_update {
      arrival {
        time: 1436918737
      }
      stop_id: "5014"
    }
    stop_time_update {
      arrival {
        time: 1436918758
      }
      stop_id: "5015"
    }
    stop_time_update {
      arrival {
        time: 1436918789
      }
      stop_id: "5016"
    }
    stop_time_update {
      arrival {
        time: 1436918803
      }
      stop_id: "5017"
    }
    vehicle {
      id: "824"
    }
    timestamp: 1436917834
  }
}
```

Psta gtfs data:

```
1209104020,19:55:00,19:55:00,4995,1,,0,0, 
1209104020,,,4996,2,,0,0, 
1209104020,,,4997,3,,0,0, 
1209104020,,,4998,4,,0,0, 
1209104020,,,4999,5,,0,0, 
1209104020,,,5000,6,,0,0, 
1209104020,,,5001,7,,0,0, 
1209104020,,,5002,8,,0,0, 
1209104020,,,5003,9,,0,0, 
1209104020,,,5004,10,,0,0, 
1209104020,,,5005,11,,0,0, 
1209104020,,,5006,12,,0,0, 
1209104020,,,9933,13,,0,0, 
1209104020,,,5007,14,,0,0, 
1209104020,,,5009,15,,0,0, 
1209104020,,,5010,16,,0,0, 
1209104020,,,5011,17,,0,0, 
1209104020,,,5012,18,,0,0, 
1209104020,,,5013,19,,0,0, 
1209104020,,,5014,20,,0,0, 
1209104020,,,5015,21,,0,0, 
1209104020,,,5016,22,,0,0, 
1209104020,,,5017,23,,0,0, 
1209104020,20:07:00,20:07:00,4809,24,,0,0, 

```

For full gtfs data: https://gist.github.com/cagryInside/b2e3603a780885636fb2

**Comment by cagryInside** [comment]
Same issue for trip_id=`1208597020`

**Comment by cagryInside** [comment]
It seems fixed for: trip_id=`1208474020` and `1208473020`, so I am closing the issue.

**Comment by barbeau** [comment]
Thanks @cagryInside!
